src/components/ListView/index.jsx

```diff
@@ -1,22 +1,60 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { useCollection } from 'react-firebase-hooks/firestore';
-import { getProducts } from 'components/Utils/firestore';
+import {
+  getProducts,
+  convertCollectionToArray,
+} from 'components/Utils/firestore';
+import { isThereCoincidence } from 'components/Utils/helpers';
 import ProductList from 'components/ProductList';
 import EmptyList from 'components/EmptyList';
 
 function ListView() {
+  const [productsFiltered, setProductsFiltered] = useState([]);
+  const [nameFilter, setNameFilter] = useState('');
   const query = getProducts();
   const [values, loading, error] = useCollection(query);
+  const handleInput = (e) => setNameFilter(e.target.value);
+
+  useEffect(() => {
+    const collectionProducts = values ? values.docs : [];
+    const arrayProducts = convertCollectionToArray(collectionProducts);
+    setProductsFiltered(arrayProducts);
```

Comment:
I would add this line to an else block. Otherwise, sometimes you are calling setProductsFiltered twice (When you go in the if block)

Reply:
Yes, you're right! 👍 , we create an else block. Thanks!